python/lsst/pipe/analysis/analysis.py

```diff
@@ -118,14 +123,20 @@ def __init__(self, catalog, func, quantityName, shortName, config, qMin=-0.2, qM
             self.fluxColumn = "flux_psf_flux"
         self.mag = -2.5*np.log10(catalog[prefix + self.fluxColumn])
 
-        self.good = np.isfinite(self.quantity) & np.isfinite(self.mag) if self.quantity is not None else None
+        self.good = (np.isfinite(self.quantity) & np.isfinite(self.mag) if self.quantity is not None
+                     else np.isfinite(self.mag))
         if errFunc is not None:
             self.good &= np.isfinite(self.quantityError)
 
-        # Don't have flags in match and overlap catalogs (already removed in the latter)
-        if ("matches" not in self.shortName and "overlap" not in self.shortName and
-                "quiver" not in self.shortName and "inputCounts" not in self.shortName):
-            for ff in set(list(self.config.flags) + flags):
+        # Don't have flags in match and overlap catalogs (already removed in the latter).
+        # Also, if selecting calib_*_used sample, do not cull on the standard
+        # self.config.flags, only on unless explicitly set in the flags
+        # variable.
+        if ("matches" not in self.shortName and "overlap" not in self.shortName and "quiver" not in
+            self.shortName and "inputCounts" not in self.shortName):
+            flagsList = flags.copy()
+            flagsList += list(self.config.flags) if self.calibUsedOnly == 0 else flagsList
```

Comment:
I don't really get what this is doing. Perhaps it could be rewritten to be clearer? Does it add `self.config.flags` to `flagsList` if `self.calibUsedOnly == 0` and if not adds `flagsList` to `flagsList`?

Reply:
I updated and added detail to the comment.